internal/pkg/shared/utils.go

```diff
@@ -37,6 +48,111 @@ func getEnvVars(defaults map[string]string, vars []corev1.EnvVar) []corev1.EnvVa
 	return allVars
 }
 
+func GenerateKeystore(alias string, pw []byte) []byte {
+	cert, derPK, err := genCert()
+	if err != nil {
+		logrus.Error(err)
+	}
+
+	var chain []keystore.Certificate
+	keyStore := keystore.KeyStore{
+		alias: &keystore.PrivateKeyEntry{
+			Entry: keystore.Entry{
+				CreationDate: time.Now(),
+			},
+			PrivKey: derPK,
+			CertChain: append(chain, keystore.Certificate{
+				Type:    "X509",
+				Content: cert,
+			}),
+		},
+	}
+
+	var b bytes.Buffer
+	err = keystore.Encode(&b, keyStore, pw)
+	if err != nil {
+		logrus.Error(err)
+	}
+
+	return b.Bytes()
+}
+
+func genCert() (cert []byte, derPK []byte, err error) {
```

Comment:
Ruben is doing something similar in the APB, please sync up with him, would be good to get confirmation that we're not missing anything and vice-versa.

Reply:
i've made improvements, most notably creating the https routes before generating certs so that the operator can use the cluster-assigned route host/url as its CN.